### PR TITLE
Ae 2457 hml superadmin perfil

### PR DESCRIPTION
**Pull request by KaynanWallen**
## Summary by CodeRabbit

* **New Features**
  * Added SearchSelect: searchable dropdown with local/async filtering, keyboard navigation, pagination/infinite scroll, variants, sizes, labels/helper/error text, loading/disabled states, controlled mode, and accessibility.

* **Documentation**
  * Storybook stories demonstrating local/async search, pagination, controlled usage, variants, sizes, states, and edge cases.

* **Tests**
  * Comprehensive test suite covering rendering, interactions, keyboard navigation, search behaviors, pagination, accessibility, and edge cases.

* **Refactor**
  * Button component updated to support forwarded refs.

* **Chores**
  * Package version bumped.

### CODE REVIEW

File: src/components/SearchSelect/SearchSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+      return styles;
+    }, [triggerRect]);
+
+    // Render options content based on state
+    const renderOptionsContent = () => {
+      if (loading) {
+        return (
+          <div className="flex items-center justify-center gap-2 p-4 text-text-500">
+            <SpinnerGap size={18} className="animate-spin" />
+            <span className="text-sm">{loadingText}</span>
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar Text

File: src/components/SearchSelect/SearchSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+                  isHighlighted && !isSelected && 'bg-background-50',
+                  option.disabled &&
+                    'opacity-50 cursor-not-allowed pointer-events-none',
+                  !option.disabled &&
+                    !isSelected &&
+                    !isHighlighted &&
+                    'hover:bg-background-50'
+                )}
+              >
+                <span className="flex-1 text-text-700">{option.label}</span>
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar text

File: src/components/SearchSelect/SearchSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+                )}
+              </div>
+            );
+          })}
+
+          {/* Loading more indicator */}
+          {loadingMore && (
+            <div className="flex items-center justify-center gap-2 p-3 text-text-500 border-t border-border-100">
+              <SpinnerGap size={16} className="animate-spin" />
+              <span className="text-xs">Carregando mais...</span>
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar text

File: src/components/SearchSelect/SearchSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,635 @@
...
+            </div>
+          )}
+
+          {/* Pagination info - only show when not filtering locally */}
+          {pagination &&
+            !loadingMore &&
+            pagination.total > 0 &&
+            !(filterLocally && searchQuery) && (
+              <div className="px-3 py-2 text-xs text-text-400 border-t border-border-100 text-center">
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar text

File: src/components/SearchSelect/SearchSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,635 @@
...
+        className="bg-secondary rounded-md border border-border-100 shadow-lg overflow-hidden flex flex-col"
+      >
+        {/* Search Input */}
+        <div className="p-2 border-b border-border-100">
+          <div className="relative">
+            <MagnifyingGlass
+              size={16}
+              className="absolute right-3 top-1/2 -translate-y-1/2 text-text-500"
+            />
+            <input
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar nosso input

File: src/components/SearchSelect/SearchSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,635 @@
...
+              size === 'medium' && 'text-md',
+              size === 'large' && 'text-lg'
+            )}
+          >
+            {label}
+          </label>
+        )}
+
+        {/* Trigger Button */}
+        <button
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar nosso button

File: src/components/SearchSelect/SearchSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,635 @@
...
+            disabled || loading
+              ? 'cursor-not-allowed opacity-50 text-text-400'
+              : 'cursor-pointer hover:bg-background-50 text-text-700'
+          )}
+        >
+          <span className={cn('truncate', !selectedLabel && 'text-text-500')}>
+            {loading ? (
+              <span className="flex items-center gap-2">
+                <SpinnerGap size={14} className="animate-spin" />
+                {loadingText}
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar text

File: src/components/SearchSelect/SearchSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,635 @@
...
+
+        {/* Helper Text / Error */}
+        {(helperText || errorMessage) && (
+          <div className="mt-1.5">
+            {helperText && !errorMessage && (
+              <p className="text-sm text-text-500">{helperText}</p>
+            )}
+            {errorMessage && (
+              <p className="text-sm text-indicator-error">{errorMessage}</p>
+            )}
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx]: Usar text

**Review by souzajr — CHANGES_REQUESTED** [review]

File: src/components/SearchSelect/SearchSelect.tsx [hunk 9]
```diff
@@ -0,0 +1,682 @@
...
+              {loadingText}
+            </Text>
+          </div>
+        );
+      }
+
+      if (filteredOptions.length === 0) {
+        return (
+          <div className="p-4 text-center text-sm text-text-500">
+            {emptyText}
```
Comment (souzajr) [inline src/components/SearchSelect/SearchSelect.tsx:459]: Text

**Review by souzajr — CHANGES_REQUESTED** [review]

**Review by souzajr — APPROVED** [review]
